### ISSUE TITLE
Crashing when using Building Gadgets

### ISSUE THREAD

**Issue by Endlessgem** [NillerMedDild/Enigmatica2Expert#1138]
**Modpack version**
1.62

**Can the issue be reproduced?**
Yes

**Logs**
https://github.com/Endlessgem/Crashreports/blob/master/crash-2019-04-22_14.35.54-client.txt

**Issue**
I can't use Builder's Wands and Building Gadgets or the game will crash.
Not sure if it's Thermal Expansion's Satchel problem.

**Comment by NillerMedDild** [comment]
Do you have a Resonant Satchel from Thermal Expansion on you?

**Comment by Frikkle** [comment]
I've been using the Building Gadget, Exchanging Gadget and Destruction Gadgets for a few weeks without any issues. Running 1.62 stock without changes.

**Comment by Endlessgem** [comment]
> Do you have a Resonant Satchel from Thermal Expansion on you?

Yes, I can use Building Gadget as usual after I delete my Resonant Satchel.

**Comment by NillerMedDild** [comment]
Thanks, I'll ask the Building Gadget folk to have another look at it.

**Comment by NillerMedDild** [comment]
It doesn't happen outside of E2E, so it's on us, and I haven't got any idea what could be modifying the Satchel unfortunately.

**Comment by MaxNeedsSnacks** [comment]
I'm also using a Resonant Satchel (with Holding IV) and I'm not getting any issues with Building Gadgets? Does it crash for you with ANY satchel, even newly spawned in ones?

**Comment by NillerMedDild** [comment]
Could you try with a new Satchel that isn't enchanted Max?

**Comment by MaxNeedsSnacks** [comment]
Sure if Twitch didn't do *this* to me right now... FeelsBadMan
![image](https://user-images.githubusercontent.com/12750501/56603441-0b78a580-6600-11e9-9e5b-c26558adcfa6.png)

**Comment by MaxNeedsSnacks** [comment]
Can't reproduce it, sorry :/

**Comment by NillerMedDild** [comment]
Huh, that's strange. I was able to reproduce it with only one attempt - Guess I need to check again :P

**Comment by MaxNeedsSnacks** [comment]
Is there more to it than just having a Building Gadget and a satchel in your inv? I can tell you that it definitely didn't recognise the TE satchel as a valid "container"

**Comment by Endlessgem** [comment]
> Is there more to it than just having a Building Gadget and a satchel in your inv? I can tell you that it definitely didn't recognise the TE satchel as a valid "container"

No, just Building Gadgets and a satchel.

**Comment by Endlessgem** [comment]
> Is there more to it than just having a Building Gadget and a satchel in your inv? I can tell you that it definitely didn't recognise the TE satchel as a valid "container"

Are you sure you're in survival mode? It's won't happen in creative mode.

**Comment by MaxNeedsSnacks** [comment]
Yup, definitely was in survival mode

**Comment by Kiefy** [comment]
I am experiencing the same crash. I don't have a Building Gadget but I do have a dank/null/.

Soon as I crafted the Resonant Satchel (Enchanted with Holding IV and XP Boost III) first chest I opened made my client crash. same (ArrayIndexOutOfBoundsException: 45) as mentioned in OP.
I suspected it was the satchel, so I placed it in a chest and left it there till today, when I decided to confirm it really was the satchel, so went back to the chest and again. same crash. I am playing on a server.

**Comment by MaxNeedsSnacks** [comment]
Huh... interesting. Which mod errored in your case, though?

**Comment by Kiefy** [comment]
Crash Log: https://pastebin.com/B8UuW8iq

**Comment by MaxNeedsSnacks** [comment]
Now *that's* interesting. You said you don't even **have** a Building Gadget?

**Comment by Kiefy** [comment]
Well, I do, in another backpack, but not in my main inv at time of crash.

**Comment by MaxNeedsSnacks** [comment]
So it wasn't in a Satchel or anything that happened to be in your inv either?

**Comment by Kiefy** [comment]
Oh actually. they are in the Cyclic Extended inv.  https://i.imgur.com/AATvpNR.png

**Comment by MaxNeedsSnacks** [comment]
Huh... I think I'm gonna leave it to MichaelHillcox here since I feel like I wouldn't contribute much to the discussion right now, but that's... interesting

**Comment by MaxNeedsSnacks** [comment]
The issue has unfortunately been marked as "unresolved" on the Building Gadgets issue tracker, but we could still just try and see whether the latest version fixes that issue

**Comment by NillerMedDild** [comment]
TL;DR for anyone being referred here:
Various mechanics that look in satchels, bags and extended inventories are treating Thermal Expansion satchels as if they have IItemHandlers registered, even though they do not.

This means that an unknown mod in Enigmatica 2: Expert is most likely assigning an IItemHandler to Thermal Expansion Satchels.

**Comment by MaxNeedsSnacks** [comment]
Common Capabilities does *something* with Thermal satchels, that may be worth looking into

**Comment by NillerMedDild** [comment]
Great find!
